docs/DeveloperDocumentation.rst

```diff
@@ -0,0 +1,104 @@
+
+* Start MySQL
+   * Open terminal enter this command ``/usr/local/mysql/bin/mysql -u your_mysql_db_username -p``. 
+     If you don't have any additional user other than root just give root as username. Press enter. Give your MySQL password.
+      
+   * create another user. Use this command. ``CREATE USER 'user_name'@'localhost' IDENTIFIED BY 'password';``
+   
+   * Grant all privileges.  ``GRANT ALL PRIVILEGES ON * . * TO 'user_name'@'localhost';``.
+   
+   * update new username and password in ``settings.xml`` file which is inside .m2 folder.
+   
+   * Type ``exit`` you will come out of MySQL terminal.
+   
+   * Now connect to MySQL with new username and password you just created ``/usr/local/mysql/bin/mysql -u your_mysql_db_username -p``.
+   
+
+* Start Apache Kafka and Zookeeper.
+   * To start Zookeeper open terminal and use this command. ``sudo /Users/your_path/zookeeper-3.3.6/bin/zkServer.sh start``. 
+   
+   * To start Kafka use this ``sudo /Users/your_path/kafka_2.11-1.1.1/bin/kafka-server-start.sh /Users/your_path/kafka_2.11   1.1.1/config/server.properties``  (there is space in between start.sh and /Users)
+
+**Perform the following steps after starting Apache Kafka and Zookeeper:**
+
+1. Clone the Citesphere repository in local from ``https://github.com/diging/citesphere``
+
+2. Import the project in eclipse → While importing make sure the import is done from the directory - ``{directory where the project is cloned}/citesphere``
+
+3. Open the ``Servers view`` (``Window → Show View → Servers``, or maybe ``Window → Show View → Other`` ...  and select ``Server``, then ``Open``) and click ``New →  Server``. 
+
+4. Now select ``Tomcat v9 Server`` and click ``Next``. Select the Tomcat installation directory and click ``Finish``.
+
+5. In the ``Servers`` view, it can be noted that the ``Tomcat v9 Server at localhost`` will be present. Double click on this. Click on the Timeouts section. Specify the start time limit as 300s and stop time limit as 15s and press ``Command + S/Ctrl + S`` to save.
+
+6. Right click on the citesphere project ``Maven → Select Maven Profiles``...Select the ``profile id`` and ensure that the id is same as the id specified in the ``settings.xml`` file in ./m2 folder. Click ``Ok``.
+
+7. Right click on ``Tomcat v9 Server at localhost`` and click ``Start``....
+
+8. Once the server is started up and running, go to ``http://localhost:8080/citesphere`` and verify if the citesphere page is loaded.
+
+9. Now, click on ``Sign Up`` at the bottom and create a new user with your credentials.
+
+10. Then logout and login with the credentials username: admin and password: admin.
+
+11. Click on ``Users`` tab. The user that was created should appear here. Add the user and click on ``Make Admin``. Now, you can logout and login back with your own credentials.
+
+12. Click on ``Connect Zotero``. It will redirect the page to the Zotero application. Click on ``Accept defaults``.
```

Comment:
This should not be "Accept defaults". I know I commented on that somewhere. Not sure if it was here or elsewhere, but this needs to be fixed.

Reply:
Actually, it must have been here because the next point says to ensure the correct key settings, so there must have been additional instructions here.